main.js

```diff
@@ -53,56 +172,118 @@ function longestWord(sen) {
  * ex. chunkArray([1, 2, 3, 4, 5, 6, 7], 3) === [[1, 2, 3],[4, 5, 6],[7]]
  * ex. chunkArray([1, 2, 3, 4, 5, 6, 7], 2) === [[1, 2],[3, 4],[5, 6],[7]]
  */
-function chunkArray(arr, len) {}
+function chunkArray(arr, len) {
+  var temp = []
+  var ans = []
+  for (x = 0; x < arr.length; x++){
+    temp.push(arr[x])
+    if ((x + 1) % len == 0 || x == arr.length - 1) {
+      ans.push(temp)
+      temp = []
+    }
+  }
+  return ans
+}
 
 
 
 /* 9. Take an array of arrays and flatten to a single array
  * ex. flattenArray([[1, 2], [3, 4], [5, 6], [7]]) should return [1, 2, 3, 4, 5, 6, 7]
  */
-function flattenArray(arrays) {}
-
-
+function flattenArray(arrays) {
+  return [].concat(...arrays)
+}
 
 /* 10. Return true if anagram and false if not
  * ex. isAnagram('elbow', 'below') === true
  */
-function isAnagram(str1, str2) {}
+function isAnagram(str1, str2) {
+  return str1.split("").sort().join("") === str2.split("").sort().join("")
+}
 
 
 
 /* 11. Change every letter of the string to the one that follows it (circularly), and capitalize the vowels
  * ex. 'hello there' === 'Ifmmp UIfsf'
  */
-function letterChanges(str) {}
+function letterChanges(str) {
+  let ans = ""
+  str.split("").forEach(x => {
+
+    let c = x.charCodeAt(0)
+    let temp = ""
+    if (c == 90) {temp = 'A'}
+    else if (c == 122) {temp = 'a'}
+    else if (c == 32) {temp = ' '}
+    else {temp = String.fromCharCode(++c)}
+    if (/[aeiou]/.test(temp)) {temp = temp.toUpperCase()}
+    ans += temp
+  });
+  return ans
+}
 
 
 
 /* 12. Return a sum of all parameters entered regardless of the amount of numbers - NO ARRAYS
  * ex. addAll(2,5,6,7) === 20
  */
-function addAll() {}
+function addAll(...arr) {
+  let sum = 0
+  arr.forEach(x => {
+    sum += x
+  });
+  return sum
+}
 
 
 
 /* 13. Pass in a number to loop up to and add all of the prime numbers. A prime number is a whole number greater than 1 whose only factors are 1 and itself
  * ex. sumAllPrimes(10) == 17
  */
-function sumAllPrimes() {}
+function sumAllPrimes(end) {
+  let sum = 0
+  function isPrime(num) {
+    for(let i = 2, s = Math.sqrt(num); i <= s; i++) {
```

Comment:
do not calculate `Math.sqrt(num)` in each iteration, pull that computation out of the loop and store in a variable.